src/lib/cfa/ntp_conf.rb

```diff
@@ -0,0 +1,387 @@
+    # return key without '[]'.
+    #
+    # For example:
+    #
+    # server 1.pool.ntp.org iburst
+    # server 2.pool.ntp.org
+    # peer 128.100.0.45
+    #
+    # In this case, key for peer entry is 'peer'.
+    # It is necessary to add '[]' to avoid possible
+    # write issues when more entries of that type
+    # are added.
+    def load
+      super
+      fix_keys
+    end
+
+    # Obtains a collection that represents the
+    # entries of the file.
+    # @return [CollectionRecord] collection to
+    #   manipulate ntp entries.
+    #
+    # The collection preserves the order of the
+    # ntp entries in the file and only contains the
+    # entries of interest (see RECORD_ENTRIES).
+    def records
+      RecordCollection.new(data)
+    end
+
+    # Obtains raw content of the file.
+    # @return [String]
+    def raw
+      PARSER.serialize(data)
+    end
+
+  private
+
+    def fix_keys
+      fix_collection_keys(data.data)
+      data.data.each do |entry|
+        if entry[:value].is_a?(AugeasTreeValue)
+          fix_collection_keys(entry[:value].tree.data)
+        end
+      end
+    end
+
+    def fix_collection_keys(entries)
+      entries.each do |entry|
+        if COLLECTION_KEYS.include?(entry[:key])
+          entry[:key] += "[]" unless entry[:key] =~ /.*\[\]$/
+        end
+      end
+    end
+
+    # class to manage ntp entries as a collection.
+    #
+    # Each element of the collection is a subclass of Record.
+    # A Record object is a wrapper of an AugeasElement.
+    class RecordCollection
+      include Enumerable
+
+      def initialize(augeas_tree)
+        @augeas_tree = augeas_tree
+      end
+
+      def each
+        record_entries.each do |augeas_element|
+          yield Record.new_from_augeas(augeas_element)
+        end
+      end
+
+      # Adds a new Record object to the collection.
+      # @param [Record] record
+      def add(record)
+        @augeas_tree.data << record.augeas
+      end
+
+      alias_method :<<, :add
+
+      # Removes a Record object from the collection.
+      # @param [Record] record
+      def delete(record)
+        matcher = Matcher.new do |k, v|
+          k == record.augeas[:key] &&
+            v == record.augeas[:value]
+        end
+        @augeas_tree.delete(matcher)
+      end
+
+      def delete_if(&block)
+        records = select(&block)
+        records.each { |record| delete(record) }
+      end
+
+      def empty?
+        count == 0
+      end
+
+      def ==(other)
+        other.to_a == to_a
+      end
+
+    private
+
+      def record_entries
+        matcher = Matcher.new do |k, _v|
+          RECORD_ENTRIES.include?(k.gsub("[]", ""))
+        end
+        @augeas_tree.select(matcher)
+      end
+    end
+
+    # Base class to represent a general ntp entry.
+    #
+    # This class is an AugeasElement wrapper. A Record
+    # has a value, and could also contains a comment and
+    # options. Its contain is stored in an AugeasElement.
```

Comment:
Really, Record class is working on augeas tree, and this is what I didn't explain correctly. I think that the examaple you propose it is correct.